src/model/AirLoopHVACOutdoorAirSystem.cpp

```diff
@@ -305,268 +353,231 @@ namespace model {
+      if (it != comps.end()) {
+        return *it;
       }
-      return result;
+
+      return boost::none;
     }
 
-    boost::optional<ModelObject> AirLoopHVACOutdoorAirSystem_Impl::oaComponent(openstudio::Handle handle) {
-      OptionalModelObject result;
+    boost::optional<ModelObject> AirLoopHVACOutdoorAirSystem_Impl::reliefComponent(openstudio::Handle handle) const {
+      auto comps = reliefComponents();
 
-      ModelObjectVector allComponents = oaComponents();
-      ModelObjectVector::iterator it;
-      for (it = allComponents.begin(); it != allComponents.end(); ++it) {
-        if (it->handle() == handle) {
-          return OptionalModelObject(*it);
-        }
+      auto it = std::find_if(comps.begin(), comps.end(), [&](const ModelObject& comp) { return (comp.handle() == handle); });
+      if (it != comps.end()) {
+        return *it;
       }
-      return result;
+
+      return boost::none;
     }
 
-    boost::optional<ModelObject> AirLoopHVACOutdoorAirSystem_Impl::reliefComponent(openstudio::Handle handle) {
-      OptionalModelObject result;
+    bool AirLoopHVACOutdoorAirSystem_Impl::addToNode(Node& node) {
 
-      ModelObjectVector allComponents = reliefComponents();
-      ModelObjectVector::iterator it;
-      for (it = allComponents.begin(); it != allComponents.end(); ++it) {
-        if (it->handle() == handle) {
-          return OptionalModelObject(*it);
-        }
+      auto airLoop_ = node.airLoopHVAC();
+
+      if (!airLoop_) {
+        return false;
+      }
+
+      if (!airLoop_->supplyComponents(this->iddObjectType()).empty()) {
+        LOG(Warn, "AirLoopHVAC " << airLoop_->nameString() << " already has an AirLoopHVACOutdoorAirSystem assigned.");
+        return false;
+      }
+
+      if (!airLoop_->supplyComponent(node.handle())) {
+        return false;
       }
-      return result;
-    }
 
-    bool AirLoopHVACOutdoorAirSystem_Impl::addToNode(Node& node) {
       Model _model = node.model();
       auto thisModelObject = getObject<ModelObject>();
 
-      if (OptionalAirLoopHVAC optionalAirLoop = node.airLoopHVAC()) {
-        AirLoopHVAC airLoop = optionalAirLoop.get();
+      if (node == airLoop_->supplyOutletNodes().front() && node.inletModelObject().get() == airLoop_->supplyInletNode()) {
+        const unsigned oldOutletPort = node.connectedObjectPort(node.inletPort()).get();
+        const unsigned oldInletPort = node.inletPort();
+        ModelObject oldSourceModelObject = node.connectedObject(node.inletPort()).get();
+        ModelObject oldTargetModelObject = node;
 
-        if (!airLoop.supplyComponents(this->iddObjectType()).empty()) {
-          return false;
-        }
+        _model.connect(std::move(oldSourceModelObject), oldOutletPort, thisModelObject, returnAirPort());
+        _model.connect(std::move(thisModelObject), mixedAirPort(), std::move(oldTargetModelObject), oldInletPort);
+        return true;
+      }
```

Comment:
Modernize the entire AirLoopHVACOutdoorAirSystem file generally speaking, trying to move as much as possible in particular.